Bootloader/Adapters/Src/flash_adapter.c

```diff
@@ -432,8 +441,9 @@ FlashAdapter_program(uint32_t address, uint8_t* buffer, uint32_t length) {
 bool
 FlashAdapter_readBytes(uint32_t address, uint8_t* buffer, uint32_t length) {

```

Comment:
It looks like the proper next step to introduce uintptr_t* everywhere we are currently using uint32_t as address storage. 
```suggestion
FlashAdapter_readBytes(uintptr_t*  address, uint8_t* buffer, uint32_t length) {

```
It will help us to be platform-agnostic and also we will get rid of suppression:
`// cppcheck-suppress misra-c2012-11.6`